### DISCUSSION TITLE
Error while executing tfc_end2end_example: Synthesis failed

### DISCUSSION THREAD

**Discussion by bferrarini** [Xilinx/finn#500] [Q&A] [answered]
Hi,

I am new to Vivado and FINN in particular.
I am working on the tutorial notebook provided within the latest FINN docker (tfc_end2end_example) to understand how to deploy BNNs on an FPGA.

I cannot pass the step, **3. Hardware Build**. I working on the notebook as is, with no changes or customization.

I get the following error:

```
Synthesis failed, no bitfile found. Check logs under /tmp/finn_dev_main/vivado_zynq_proj_2l5uh0qe
```

I checked the vivado.log file I and I found a description of the problem:
```
# set_property ip_repo_paths [concat [get_property ip_repo_paths [current_project]] [list /tmp/finn_dev_main/code_gen_ipgen_StreamingDataflowPartition_0_IODMA_0_tglba_xo/project_StreamingDataflowPartition_0_IODMA_0/sol1/impl/ip /tmp/finn_dev_main/vivado_stitch_proj_eslvsrpf/ip]] [current_project]
# update_ip_catalog -rebuild -scan_changes
INFO: [IP_Flow 19-234] Refreshing IP repositories
INFO: [IP_Flow 19-1700] Loaded user IP repository '/tmp/finn_dev_main/code_gen_ipgen_StreamingDataflowPartition_0_IODMA_0_tglba_xo/project_StreamingDataflowPartition_0_IODMA_0/sol1/impl/ip'.
WARNING: [IP_Flow 19-2248] Failed to load user IP repository '/tmp/finn_dev_main/vivado_stitch_proj_eslvsrpf/ip'; Can't find the specified path.
If this directory should no longer be in your list of user repositories, go to the IP Settings dialog and remove it.
# create_bd_cell -type ip -vlnv xilinx_finn:finn:StreamingDataflowPartition_0:1.0 idma0
ERROR: [BD 5-390] IP definition not found for VLNV: xilinx_finn:finn:StreamingDataflowPartition_0:1.0 
ERROR: [Common 17-39] 'create_bd_cell' failed due to earlier errors.
```

**My environment:**

- Ubuntu 20.04
- Vivado 2020.1 (I tried some others: 2018.3 and 2019.1. Newer Vivado does not have vivado_hls script)
- Minized Board

I don't know if I am missing something on the FINN's requirements (e.g. Vivado version).

In the getting started guide [(https://finn.readthedocs.io/en/v0.2b/getting_started.html)](https://finn.readthedocs.io/en/v0.2b/getting_started.html) and FINN-notebook I cannot find any useful indication on this.

Can anyone give me some hints?

Regards

**Comment by fpjentzsch** [comment, marked as answer]
Hi, did you apply the patch for Vivado's Y2K22 bug (see https://github.com/Xilinx/finn/discussions/483)? This breaks IP export after 01.01.2022.

If that's not the reason, you'll need to examine the logs for the HLS IP that failed to generate under "/tmp/finn_dev_main/vivado_stitch_proj_eslvsrpf/..." in this case.

**Reply by bferrarini** [reply]
The missing patch was the problem.

Many Thanks for your help :)
